packages/viewtoggle/src/react/index.js

```diff
@@ -19,138 +18,119 @@ const styles = {
   pillBgSpacer: props => glamor.css(css['.psds-viewtoggle__option-bg__spacer'])
 }
 
-const OptionButton = props => (
-  <button
-    {...styles.optionButton(props)}
-    {...filterReactProps(props, { tagName: 'button' })}
-  />
-)
+const ViewToggle = React.forwardRef(({ onSelect, ...props }, forwardedRef) => {
+  const ref = forwardedRef || React.useRef()
 
-class Option extends React.Component {
-  constructor(props) {
-    super(props)
-    this.handleClick = this.handleClick.bind(this)
-  }
+  const initialIndex = findActiveIndex(props.children)
+  const [activeIndex, setActiveIndex] = React.useState(initialIndex)
 
-  componentDidMount() {
-    if (this.props.active) setTimeout(_ => this.handleClick(), 0)
-  }
+  React.useEffect(() => {
+    const index = findActiveIndex(props.children)
+    setActiveIndex(index)
+  }, [props.children])
 
-  handleClick(evt) {
-    this.props._onSelect(this.props._i, evt)
+  function handleSelect(evt, index) {
+    setActiveIndex(index)
+
+    // TODO: update this prop callback so that Event is the first param.
```

Comment:
Should we do this TODO while we're already introing a breaking change/major version?